apollo/integrations/snowflake/snowflake_proxy_client.py

```diff
@@ -48,3 +50,27 @@ def get_error_extra_attributes(self, error: Exception) -> Optional[Dict]:
                 "sqlstate": error.sqlstate,
             }
         return super().get_error_extra_attributes(error)
+
+    def execute_sql_query(
+        self, sql_query: str, max_results: int, query_timeout: int
+    ) -> AgentExecuteSqlQueryResponse:
+        """
+        Execute a SQL query synchronously and collect results.
+        """
+        with self._connection.cursor() as cursor:
+            cursor.execute(sql_query, timeout=query_timeout)
+            results = cursor.fetchmany(max_results + 1)
+            if len(results) > max_results:
+                is_partial = True
+                results = results[:-1]
+            else:
+                is_partial = False
+
+            description = cursor.description or []
+
+            return AgentExecuteSqlQueryResponse(
+                number_of_rows_fetched=len(results),
+                field_names=[field[0] for field in description],
+                rows=results,
+                is_partial=is_partial,
```

Comment:
Might change this schema a little. Now, it mimics what the data collector is returning on the execute SQL query, so we don't need a different implementation depending if we are going against an agent or a legacy DC.